framework/src/governance/lib/datazone/glue-schema-registry-version.ts

```diff
@@ -0,0 +1,137 @@
+import { CustomResource, Duration, RemovalPolicy } from 'aws-cdk-lib';
+import {
+  Effect,
+  IRole,
+  ManagedPolicy,
+  PolicyDocument,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from 'aws-cdk-lib/aws-iam';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { ILogGroup } from 'aws-cdk-lib/aws-logs';
+import { Construct } from 'constructs';
+import { GlueSchemaRegistryVersionProps } from './glue-schema-registry-version-props';
+import { Context, TrackedConstruct, TrackedConstructProps } from '../../../utils';
+import { DsfProvider } from '../../../utils/lib/dsf-provider';
+
+export class GlueSchemaRegistryVersion extends TrackedConstruct {
+  private readonly removalPolicy: RemovalPolicy;
+
+  readonly createLogGroup: ILogGroup;
+  readonly createFunction: IFunction;
+  readonly createRole: IRole;
+  readonly statusLogGroup: ILogGroup;
+  readonly statusFunction: IFunction;
+  readonly statusRole: IRole;
+  readonly serviceToken: string;
+  // Exposed properties
+  readonly schemaArn?: string;
+  readonly registryName?: string;
+  readonly schemaName?: string | undefined;
+  readonly schemaVersionNumber?: number;
+  readonly latestVersion?: boolean;
+
+  // Response properties
+  readonly schemaVersionId: string;
+  readonly schemaDefinition: string;
+  readonly dataFormat: string;
+  readonly versionNumber: string;
+  readonly status: string;
+  readonly createdTime: string;
+  constructor(scope: Construct, id: string, props: GlueSchemaRegistryVersionProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: GlueSchemaRegistryVersion.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+
+    // Validate props
+    if (props.schemaArn && props.registryName) {
+      throw new Error('You cannot specify both schemaRegistryArn and registryName. Please provide only one.');
+    }
+
+    if (!props.schemaArn && !props.registryName) {
+      throw new Error('You must specify either schemaRegistryArn or registryName.');
+    }
+
+
+    if (props.schemaVersionNumber === undefined && props.latestVersion === undefined) {
+      throw new Error('Either schemaVersionNumber or latestVersion must be provided.');
+    }
+
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props?.removalPolicy);
+    const handlerRole = new Role(this, 'HandlerRole', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
```

Comment:
@FranMorilloAWS  please use a policy created within the DSF and not the basic execution role.